### PR TITLE
Fix off-by-one in pem_find_pattern rejecting PEM without a trailing newline

### PR DESCRIPTION
**Pull request by gasbytes**
# Description

Changed both scans loops in pem_find_pattern to <= so PEM ending at the footer's final dash parses.

# Testing

Updated the general test test_wolfssl_pem_read to test this edge case too.
# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by gasbytes** [comment]
Jenkins retest this please (logs lost)

**Comment by gasbytes** [comment]
Jenkins retest this please (timeout)

**Comment by gasbytes** [comment]
Jenkins retest this please (timeout)

**Comment by philljj** [comment]
skoll locally ran clean, no comments.

File: src/pk.c [hunk 1]
```diff
@@ -6847,7 +6847,7 @@ static void pem_find_pattern(char* pem, int pemLen, int idx, const char* prefix,
         }
     }
     /* Find postfix part. */
-    for (; idx < pemLen - postfixLen; idx++) {
+    for (; idx <= pemLen - postfixLen; idx++) {
```
[the remark this answers is not in the document]
Comment (philljj) [reply src/pk.c:6850]: this feels like a false positive but not sure

**Review by philljj — CHANGES_REQUESTED** [review]
Please review if Fenrir suggestion is worth incorporating. Feels marginal

Looks good otherwise.

File: src/pk.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (gasbytes) [reply src/pk.c:6850]: seems legit imo, should be addressed now.

**Comment by gasbytes** [comment]
Jenkins retest this please

**Comment by gasbytes** [comment]
Jenkins retest this please

**Review by philljj — APPROVED** [review]
